instruments/thorlabs/_abstract.py

```diff
@@ -72,7 +72,7 @@ def querypacket(self, packet, expect=None, timeout=None, expect_data_len=None):
         """
         t_start = time.time()
 
-        if timeout:
+        if timeout is not None:
```

Comment:
Setting the timeout to 0 for the tests to speed things up. This might not ever be done in real life, but is a possibility, and in that case the routine would fail. Now checking if timeout is `None`, which is what it should be if not set.

Reply:
:thinking: `bool(None)` evaluates to `False`, so I'm not sure what case we're catching here

Reply:
Some more content: 
This part of the `querypacket` routine, which initializes as following: 

```python
def querypacket(self, packet, expect=None, timeout=None, expect_data_len=None):
    ...
```

So if the keyword argument `timeout` is not specified, it is `None`. If it is not `None`, seconds are assumed for the units. To test setting with a set timeout I call the routine with a timeout of zero. However, if `timeout = 0` then `if timetout` comes out false. Thus I figured that the original intention was to catch the default setting, i.e., `timeout = None` and am checking for that. Could go the other way and replace that with 

```python
if isinstance(timeout, int) or isinstance(timeout, float)
```

Reply:
> However, if timeout = 0 then if timetout comes out false.

Gotcha